timemachine/cpp/src/context.cu

```diff
@@ -263,7 +193,8 @@ std::array<std::vector<double>, 2> Context::multiple_steps_local(
         int num_row_indices = p_num_selected.data[0];
         int num_col_indices = N_ - num_row_indices;
         if (num_row_indices == 0 || num_col_indices == 0) {
-            throw std::runtime_error("local md no longer stable, check system");
+            throw std::runtime_error(
+                "entire system selected; reduce radius, increase k, check system or use Context::multiple_steps");
```

Comment:
Missed this in previous PR, but is it more appropriate for this to be a warning rather than an error? (Since there's a nonzero chance of this happening under normal operation, and selecting the whole system would cause no problems in principle.)

If we need to throw an error here because it is difficult to throw an error closer to the assumption this would violate (which may be independent of local MD, maybe in the neighborlist?), please expand this error message accordingly.

Reply:
We could have it fall back on global MD in this case.

Though in the case of falling back to global do we still want the restraint?

Reply:
>We could have it fall back on global MD in this case.

Possibly -- but still warn of possibly unintended configuration?

>Though in the case of falling back to global do we still want the restraint?

Yes.

Let's say we have a nonzero chance of selecting each particle in the system independently (which is true with all settings of the currently implemented variant). Then on any given iteration, there's also a nonzero chance of selecting all the particles at once. We might have a branch that switches to a more efficient implementation on those iterations, but omitting the restraint when that branch is taken would not be correct.

There are special cases where omitting the restraint would be equivalent to including the restraint. (E.g. if the flat-bottom radius is very large, so that there's always a 100% chance of selecting every particle.) But in general, I think we want the implementation to be equivalent to including a restraint function that is [consistent with the selection probability function](https://github.com/proteneer/timemachine/blob/2779aef8c3a30f2e28b1c0aae9f77cf27f816f55/timemachine/md/local_resampling.py#L23-L25).

Reply:
^ Non-blocking -- can be considered further in a later PR 👍